src/Core/Auth/Services/Implementations/TwoFactorEmailService.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System.ComponentModel.DataAnnotations;
+using System.Reflection;
+using Bit.Core.Auth.Enums;
+using Bit.Core.Context;
+using Bit.Core.Entities;
+using Bit.Core.Services;
+using Bit.Core.Utilities;
+using Core.Auth.Enums;
+using Microsoft.AspNetCore.Identity;
+
+namespace Bit.Core.Auth.Services;
+
+public class TwoFactorEmailService : ITwoFactorEmailService
+{
+    private readonly ICurrentContext _currentContext;
+    private readonly UserManager<User> _userManager;
+    private readonly IMailService _mailService;
+
+    public TwoFactorEmailService(
+        ICurrentContext currentContext,
+        IMailService mailService,
+        UserManager<User> userManager
+    )
+    {
+        _currentContext = currentContext;
+        _userManager = userManager;
+        _mailService = mailService;
+    }
+
+    public async Task SendTwoFactorEmailAsync(User user)
```

Comment:
There is only one difference between SendTwoFactorEmailAsync and SendTwoFactorSetupEmailAsync: the enum passed to the `MailService`.

I would suggest creating a private method that captures the redundancies, and accepts the enum and user as the input.

```suggestion
    public async Task SendTwoFactorEmailAsync(User user)
    {
      await VerifyAndSendTwoFactorEmailAsync(user, TwoFactorEmailPurpose.Login);
    }
```